astropy/units/quantity.py

```diff
@@ -861,6 +861,43 @@ def __ne__(self, other):
         except TypeError:
             return NotImplemented
 
+    # Unit conversion operator (|).
+    def __or__(self, other):
+        try:
+            other = Unit(other, parse_strict='silent')
+        except UnitTypeError:
+            return NotImplemented
+
+        return self.__class__(self, other, copy=False, subok=True)
+
+    def __ior__(self, other):
+        try:
+            other = Unit(other, parse_strict='silent')
+        except UnitTypeError:
+            return NotImplemented
```

Comment:
Doesn't the presence of `NotImplemented` mean it will try to fall back on the "normal" meaning of ``|`` (i.e., "or")?  Do we actually *want* that?  It feels like it might be safer to instead explicitly block that by throwing a `TypeError` or something like that?

Reply:
(Note that this is pretty minor as it's sort of "future-proofing" - it looks like the ``|`` operator raises the standard `TypeError` anyway for most cases I tried)

Reply:
`NotImplemented` means that `other` gets a chance - just in case someone implements their own unit class or so that can, say, interact with `Quantity` or perhaps even with any `ndarray` class, but which we do not know about. In the normal case that `other` doesn't know what to do either, one gets a `TypeError`.

Reply:
Right, I'm thinking we might want to *disallow* that as it might cause more confusion than it's worth.  But I see your point that if this is a user/subclass situation, there might be some use cases, so it's probably not protecting that much.